src/components/include/utils/threads/thread.h

```diff
@@ -172,7 +173,11 @@ class Thread {
    */
   void stop();
 
-  void join();
+  /**
+   * @brief Starts join procedure.
+   * @param[in] force_stop If true stop() is called first otherwise is skipped.
+   */
```

Comment:
@orhan-mehmedov,  I truly recommend to fully describe behavious  with true and false flags.